clang/lib/AST/Type.cpp

```diff
@@ -4205,6 +4205,7 @@ SubstTemplateTypeParmType::SubstTemplateTypeParmType(
 
   SubstTemplateTypeParmTypeBits.Index = Index;
   SubstTemplateTypeParmTypeBits.PackIndex = PackIndex ? *PackIndex + 1 : 0;
+  SubstTemplateTypeParmTypeBits.ExpandPacksInPlace = ExpandPacksInPlace;
```

Comment:
ExpandPacksInPlace should only ever be true if there is a PackIndex. Can we enforce that here, and other places which are relevant, like profiling?